actix-verify/src/utils.rs

```diff
@@ -0,0 +1,157 @@
+use actix_web::HttpRequest;
+
+use super::*;
+
+// verification function
+pub async fn verify_sig(request: &HttpRequest) -> bool {
+    // Getting the public from headers
+    let hm_key = match request.headers().get("HM-KEY") {
+        Some(pk) => {
+            info!("{:?}", pk);
+            pk.to_str().unwrap()
+        }
+        None => {
+            error!("Public Key not Found");
+            return false;
+        }
+    };
+
+    // Taking the signature from headers
+    let hm_sign = match request.headers().get("HM-SIGN") {
+        Some(sign) => sign.to_str().unwrap(),
+        None => {
+            error!("Missing Signature");
+            return false;
+        }
+    };
+
+    // Decoding the timestamp from headers.
+    let ts = match request.headers().get("HM-TS") {
+        Some(timestamp) => timestamp.to_str().unwrap().parse::<u64>().unwrap(),
+        None => {
+            error!("Missing timestamp in Headers");
+            return false;
+        }
+    };
+
+    // Extracting the public
+    let public_key = PublicKey::from_slice(&hex::decode(hm_key).unwrap()).unwrap();
+    info!("Public key  found {}", public_key);
+    // Extract signature from string
+    let signature = Signature::from_compact(&base64::decode(hm_sign).unwrap());
+    // let s :Signature;
+    match signature {
+        Ok(x) => println!("{:?}", x),
+        // s =x;},
+        Err(e) => {
+            println!("{:?}", e)
```

Comment:
You don't exit here, later you unwrap the signature. So we will panic in this case.